event-stream-processing/src/node-css-parser.test.ts

```diff
@@ -0,0 +1,149 @@
+import { NodeCssParser } from './node-css-parser';
+
+test('basic', () => {
+  const filter = new NodeCssParser();
+
+  /* eslint-disable */
+
+  // record from fluent-bit looks like this:
+  /*
+
+  [
+    1
+  ] app.http: [
+    1626304625.036902482,
+    {
+      "statusCode"=>200,
+      "userAgent"=>"Mozilla/5.0",
+      "path"=>"/api/v1/forms",
+      "message"=>"GET /api/v1/forms 200 624ms",
+      "logFilePath"=>"/var/log/app.log",
+      "logFileOffset"=>759,
+      "hostname"=>"chefs-app-123",
+      "timestamp"=>"2021-07-14T23:17:05.036Z",
+      "product"=>"chefs",
+      "namespace"=>"123-dev",
+      "azp"=>"chefs-frontend",
+      "level"=>"http",
+      "contentLength"=>"323",
+      "agent"=>{
+        "type"=>"fluentbit"
+      },
+      "responseTime"=>624,
+      "httpVersion"=>"1.1",
+      "event"=>{
+        "type"=>"http",
+        "module"=>"node.css",
+        "kind"=>"event",
+        "dataset"=>"node.css.http"
+      },
+      "ip"=>"::ffff:10.97.8.1",
+      "log"=>"{"responseTime":624,"azp":"chefs-frontend","contentLength":"323","httpVersion":"1.1","ip":": :ffff: 10.97.8.1","method":"GET","path":"/api/v1/forms","statusCode":200,"userAgent":"Mozilla/5.0","level":"http","message":"GET /api/v1/forms 200 624ms","timestamp":"2021-07-14T23: 17: 05.036Z"}",
+      "method"=>"GET"
+    }
+  ]
+  */
+  /* eslint-enable */
+
+
+  // --- the test record object for parsing
+  const record = {
+    'agent': {
+      'type': 'fluentbit',
+    },
+    'azp': 'chefs-frontend',
+    'contentLength': '323',
+    'event': {
+      'type': 'http',
+      'module': 'node.css',
+      'kind': 'event',
+      'dataset': 'node.css.http',
+    },
+    'hostname': 'chefs-app-123',
+    'httpVersion': '1.1',
+    'ip': '::ffff:10.97.8.1',
+    'level': 'http',
+    'log': {
+      'responseTime': 624,
+      'azp': 'chefs-frontend',
+      'contentLength': '323',
+      'httpVersion': '1.1',
+      'ip': ': :ffff: 10.97.8.1',
```

Comment:
this a weird format to me. It seems like concatenation of ipv6 + ': ' + ipv4 (like the `.ip` field above), but that first `: ` (with a space) is throwing me off.  Do you know how `log.ip` differs from `ip`?

Reply:
well spotted.. this extra space was caused by me doing 'format document' in the VS Code editor.
now passing 'client' as ip address

Reply:
missunderstood.. no getting either ipv4 or ipv6 ip address